### PR TITLE
Fix #753, clarification this version of the specification

### LINKED ISSUE #753: "This version of the specification"
This review is based on the spec from Sept. 1st, 2023.

The spec uses very often "This version of the specification". Most of the time it is to impose restrictions on bitstream or parsers. We have to be careful here.

1. what is a version? Does it refer to the header of the spec that currently says "This version:"? Every time we'll make a change to the spec, we'll publish a new version (1.1, 1.1.1, 1.2, 1.2.1, ...), see AV1-ISOBMFF for example. What will happen the wording "in this version of the specification"? Will we have to change it to "in this and previous versions"?

2. Note that the version is not coded in the bitstream (by design).

3. I think we should do a case-by-case review of these sentences and rephrase them, as suggested below

> In this version of the specification, the term IA OBU can be used interchangeably with OBU.

into:
> In this specification, the term IA OBU can be used interchangeably with OBU.

a. obu_extension_flag 
> This flag SHALL be set to 0 for this version of the specification. An OBU parser that is conformant with this version of the specification SHOULD ignore the [extension_header_bytes](https://aomediacodec.github.io/iamf/#extension_header_bytes).

This is saying that a bitstream compliant to this version (not identifiable from the bistream) is invalid. Given the next sentence, I'm not sure why the restriction exists. Also, if a next version actually starts defining bytes for extension headers, that sentence would have to change. I would suggest deleting the first sentence.

b. extra bytes
>The [obu_size](https://aomediacodec.github.io/iamf/#obu_size) MAY be greater than the size needed to represent the OBU syntax defined in this version of the specification, for example, to represent new syntax defined in a future version of the specification. Parsers compliant with this version of the specification SHOULD ignore these bytes.

I don't think the use of "defined in this version of the specification" is necessary. It should always be true, even in the future. The size indicates all the bytes. I would rephrase the entire paragraph:
>The [obu_size](https://aomediacodec.github.io/iamf/#obu_size) MAY be greater than the size needed to represent the OBU syntax. Parsers SHOULD ignore bytes past the OBU syntax that they recognize.

c. extension_header_bytes 
> extension_header_bytes indicates the byte representations of the syntaxes of the extension header. Parsers compliant with this version of the specification SHOULD ignore these bytes.

Rephrase as:
> extension_header_bytes indicates the byte representations of the syntaxes of the extension header. Parsers that don't understand these bytes SHOULD ignore them.

d. Reserved OBUs
Rephrase
> Reserved OBUs SHOULD be ignored by parsers compliant with this version of the specification. Future versions of the specification MAY define syntax and semantics for these [Reserved OBU](https://aomediacodec.github.io/iamf/#reserved-obu)s that would only be supported by parsers compliant with these future versions.

into
> Parsers SHOULD ignore Reserved OBUs. 
NOTE: Future versions of the specification MAY define syntax and semantics for an `obu_type` value, making it no longer a [Reserved OBU](https://aomediacodec.github.io/iamf/#reserved-obu)s for those parsers compliant with these future versions.

e. primary profile
I would suggest changing:
> primary_profile indicates the primary profile that this [IA Sequence](https://aomediacodec.github.io/iamf/#ia-sequence) complies with. Parsers compliant with this version of the specification SHOULD discard the [IA Sequence](https://aomediacodec.github.io/iamf/#ia-sequence) if they do not support the value indicated here.

into:
> primary_profile indicates the primary profile that this [IA Sequence](https://aomediacodec.github.io/iamf/#ia-sequence) complies with. Parsers SHOULD discard the [IA Sequence](https://aomediacodec.github.io/iamf/#ia-sequence) if they do not support the value indicated here.

f. codec_id
Change
> For this version of the specification, it SHALL be set to one of the four [codec_id](https://aomediacodec.github.io/iamf/#codec_id) values defined below:

into:
> This specification supports the following four [codec_id](https://aomediacodec.github.io/iamf/#codec_id) values defined below:

and change:
> Parsers compliant with this version of the specification SHOULD ignore [Codec Config OBU](https://aomediacodec.github.io/iamf/#codec-config-obu)s with an unknown [codec_id](https://aomediacodec.github.io/iamf/#codec_id).

into:
> Parsers SHOULD ignore [Codec Config OBU](https://aomediacodec.github.io/iamf/#codec-config-obu)s with a [codec_id](https://aomediacodec.github.io/iamf/#codec_id) that they don't support.

g. audio_element_type
Change
> Parsers compliant with this version of the specification SHOULD ignore [Audio Element OBU](https://aomediacodec.github.io/iamf/#audio-element-obu)s with a reserved [audio_element_type](https://aomediacodec.github.io/iamf/#audio_element_type).

into
> Parsers SHOULD ignore [Audio Element OBU](https://aomediacodec.github.io/iamf/#audio-element-obu)s with an  [audio_element_type](https://aomediacodec.github.io/iamf/#audio_element_type) that they don't support.

h. codec_config_id
Change
> Parsers compliant with this version of the specification SHOULD ignore [Audio Element OBU](https://aomediacodec.github.io/iamf/#audio-element-obu)s with a [codec_config_id](https://aomediacodec.github.io/iamf/#audio_element_obu-codec_config_id) identifying an unknown [codec_id](https://aomediacodec.github.io/iamf/#codec_id).

into:
> Parsers SHOULD ignore [Audio Element OBU](https://aomediacodec.github.io/iamf/#audio-element-obu)s with a [codec_config_id](https://aomediacodec.github.io/iamf/#audio_element_obu-codec_config_id) identifying a [codec_id](https://aomediacodec.github.io/iamf/#codec_id) that they don't support.

i. num_parameters
Regarding:
> * When [audio_element_type](https://aomediacodec.github.io/iamf/#audio_element_type) = 0, this field SHALL be set to 0, 1, or 2 for this version of the specification.
> * When [audio_element_type](https://aomediacodec.github.io/iamf/#audio_element_type) = 1, this field SHALL be set to 0 for this version of the specification.
> * Parsers compliant with this version of the specification SHALL be able to parse any value of [num_parameters](https://aomediacodec.github.io/iamf/#num_parameters).

I wonder if the first 2 bullet points shouldn't be in the profiles section, or if "for this version of the specification" shouldn't be removed as it is probably valid for all future bitstreams.

I would rephrase the 3rd sentence:
> Parsers SHALL support any value of [num_parameters](https://aomediacodec.github.io/iamf/#num_parameters).

j. param_definition_type
Change:
> All parameter definition types described in this version of the specification are listed in the table below, along with their associated parameter definitions.

into:
> The parameter definition types are listed in the table below, along with their associated parameter definitions.

and:
> An OBU parser that is conformant with this version of the specification SHALL be able to parse [param_definition_type](https://aomediacodec.github.io/iamf/#param_definition_type) = P (where P > 2) and [param_definition_size](https://aomediacodec.github.io/iamf/#param_definition_size). The OBU Parser SHOULD ignore the bytes indicated by [param_definition_size](https://aomediacodec.github.io/iamf/#param_definition_size).

into:
> An OBU parser SHALL be able to parse [param_definition_type](https://aomediacodec.github.io/iamf/#param_definition_type) = P (where P > 2) and [param_definition_size](https://aomediacodec.github.io/iamf/#param_definition_size). The OBU Parser SHOULD ignore the bytes indicated by [param_definition_size](https://aomediacodec.github.io/iamf/#param_definition_size) that it does not recognize.

k. param_definition_bytes 
> param_definition_bytes represents reserved bytes for future use when new [param_definition_type](https://aomediacodec.github.io/iamf/#param_definition_type) values are defined. Parsers compliant with this version of the specification SHOULD ignore these bytes.

into:
> param_definition_bytes represents reserved bytes for future use when new [param_definition_type](https://aomediacodec.github.io/iamf/#param_definition_type) values are defined. Parsers SHOULD ignore these bytes when they don't understand the parameter definition.

l. audio_element_config_bytes
> audio_element_config_bytes represents reserved bytes for future use when new [audio_element_type](https://aomediacodec.github.io/iamf/#audio_element_type) values are defined. Parsers compliant with this version of the specification SHOULD ignore these bytes.
into:
> audio_element_config_bytes represents reserved bytes for future use when new [audio_element_type](https://aomediacodec.github.io/iamf/#audio_element_type) values are defined. Parsers SHOULD ignore these bytes when they don't support a particular configuration.

[To be continued, in the same spirit]

### PR DESCRIPTION
**Pull request by sunghee-hwang**
***
<a href="https://pr-preview.s3.amazonaws.com/AOMediaCodec/iamf/pull/756.html" title="Last updated on Sep 11, 2023, 12:43 AM UTC (109b07c)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/AOMediaCodec/iamf/756/42194f0...109b07c.html" title="Last updated on Sep 11, 2023, 12:43 AM UTC (109b07c)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -796,7 +785,7 @@ In this parameter definition,
 
 <dfn noexport>audio_element_config_size</dfn> indicates the size in bytes of [=audio_element_config_bytes=].
 
-<dfn noexport>audio_element_config_bytes</dfn> represents reserved bytes for future use when new [=audio_element_type=] values are defined. Parsers compliant with this version of the specification SHOULD ignore these bytes.
+<dfn noexport>audio_element_config_bytes</dfn> represents reserved bytes for future use when new [=audio_element_type=] values are defined. Parsers SHOULD ignore these bytes when they don't support a particular configuration.
```
Comment (sunghee-hwang) [inline index.bs:788]:
```suggestion
<dfn noexport>audio_element_config_bytes</dfn> represents reserved bytes for future use when new [=audio_element_type=] values are defined. Parsers SHOULD ignore these bytes when they don't recognize a particular configuration.
```

File: index.bs [hunk 2]
```diff
@@ -1501,13 +1490,13 @@ class ParameterBlockOBU() {
...
+<dfn noexport for="parameter_block_obu">parameter_id</dfn> indicates the identifier for a [=Parameter Substream=] which this [=Parameter Block OBU=] refers to. If no [=Audio Element OBU=]s or [=Mix Presentation OBU=]s refer to this [=parameter_block_obu/parameter_id=], parsers SHOULD ignore [=Parameter Block OBU=]s with this identifier.
 
 <dfn noexport>get_param_definition()</dfn> is a run-time function to get the [=param_definition_type=] and [=param_definition_mode=] from the [=Audio Element OBU=] or [=Mix Presentation OBU=] that references this [=parameter_block_obu/parameter_id=].
 
 If [=param_definition_mode=] = 0, this function additionally gets the following fields from the same [=Audio Element OBU=] or [=Mix Presentation OBU=]: [=ParamDefinition/duration=], [=ParamDefinition/num_subblocks=], [=ParamDefinition/constant_subblock_duration=], and [=ParamDefinition/subblock_duration=].
 
-When it gets an unknown [=param_definition_type=], parsers compliant with this version of the specification SHOULD ignore the [=Parameter Block OBU=]. 
+Parsers SHOULD ignore the [=Parameter Block OBU=] with a [=param_definition_type=] that they don't support.
```
Comment (sunghee-hwang) [inline index.bs:1499]:
```suggestion
Parsers SHOULD ignore the [=Parameter Block OBU=] with a [=param_definition_type=] that they don't recognize. 
```

File: index.bs [hunk 3]
```diff
@@ -943,7 +937,7 @@ The relationship among all [=Channel Group=]s for the given scalable channel aud
 
 <dfn noexport>channel_audio_layer_config</dfn> is an instance of the [=ChannelAudioLayerConfig()=] class, which provides the i-th [=Channel Group=]'s configuration, where i is the layer index provided as input argument to this instance of the [=ChannelAudioLayerConfig()=] class.
 
-<dfn noexport>loudspeaker_layout</dfn> indicates the channel layout to be reconstructed from the precedent [=Channel Group=]s and current [=Channel Group=]. When a reserved value for [=loudspeaker_layout=] is used, parsers compliant with this version of the specification SHOULD skip the [=channel_audio_layer_config=] for that layer and all subsequent ones, if any.
+<dfn noexport>loudspeaker_layout</dfn> indicates the channel layout to be reconstructed from the precedent [=Channel Group=]s and current [=Channel Group=]. If parsers do not recognize a [=loudspeaker_layout=] for a particular layer, they SHOULD skip the [=channel_audio_layer_config=] for that layer, all subsequent layers.
```
Comment (felicialim) [inline index.bs:940]:
```suggestion
<dfn noexport>loudspeaker_layout</dfn> indicates the channel layout to be reconstructed from the precedent [=Channel Group=]s and current [=Channel Group=]. If parsers do not recognize a [=loudspeaker_layout=] for a particular layer, they SHOULD skip the [=channel_audio_layer_config=] for that layer and all subsequent layers.
```

**Review by felicialim — APPROVED** [review]
Thanks for making these changes!
